x-pack/legacy/plugins/triggers_actions_ui/np_ready/public/application/sections/alert_add/alert_reducer.test.ts

```diff
@@ -0,0 +1,163 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { alertReducer } from './alert_reducer';
+import { Alert } from '../../../types';
+
+describe('alert reducer', () => {
+  let initialAlert: Alert;
+  beforeAll(() => {
+    initialAlert = ({
+      params: {},
+      consumer: 'alerting',
+      alertTypeId: null,
+      schedule: {
+        interval: '1m',
+      },
+      actions: [],
+      tags: [],
+    } as unknown) as Alert;
```

Comment:
We should avoid ((... as unknown) as Alert) in tests as it likely hides mistakes in the typing that now won't get caught by the test.
This caused a few mistakes in 7.6, so we should really try to use sounds types in tests as we ll as production code

Reply:
Looking closer it looks like the first test actually relies on this not being a valid alert (Name is undefined?) but the others don't , so I think the other tests should receive a full valid alert and the first one should explicitly be handed an alert where it's clear that the name field is intentionally omitted, otherwise we might finbd it hard to maintain this code in the future when the context is forgotten. :)